api/src/main/java/org/aibles/java/api/crud/service/impl/ProductServiceImpl.java

```diff
@@ -42,19 +42,19 @@ public List<ProductReponse> getAllProducts() {
     }
 
     @Override
-    public ProductReponse getProductById(Long id) throws ProductNotFoundException {
+    public ProductReponse getProductById(Long id) {
         log.info("GET PRODUCT WITH ID: {}",id);
         Optional<ProductEntity> optionalProduct = productRepository.findById(id);
         if (optionalProduct.isPresent()) {
             ProductEntity product = optionalProduct.get();
             return new ProductReponse(product.getId(), product.getProduct_name(), product.getPrice());
         } else {
-            throw new ProductNotFoundException("Product not found with id: " + id);
+            throw new RuntimeException("Product not found with id: " + id);
```

Comment:
đoạn này em nên trả về ProductNotFoundException. vì ở bên handler em đang trả bắt nó để xử lí
![image](https://github.com/hoangducdat/product-manager/assets/69207716/0e6f46e0-ff65-47f5-958c-fe7bdce2b9e7)

Reply:
Em có thể code đoạn này thành
if (!optionalProduct.isPresent()) {
         throw new ProductNotFoundException("Product not found with id: " + id);
        }
        
  ProductEntity product = optionalProduct.get();
  return new ProductReponse(product.getId(), product.getProduct_name(), product.getPrice());